Flaky: GatewayAuthenticationNoneIT fails with Docker Hub pull timeout in CI

### <!-- Component -->
<!-- Zeebe- -->

### <!-- Affected version -->
<!-- -8.10 -->

### <!-- Severity -->
<!-- Low- -->

### <!-- Likelihood -->
<!-- High_ -->

### Description
`io.camunda.zeebe.it.shared.gateway.GatewayAuthenticationNoneIT` fails intermittently in CI when
Testcontainers tries to pull `camunda/identity:8.10.0-alpha2` from Docker Hub. The pull stalls
silently for 2 minutes before Testcontainers' Awaitility window expires and throws
`ContainerFetchException`. The test is then retried 1–3 times and eventually passes.

This is a CI-only issue. It does not affect any deployed product environment.

### Steps to reproduce
1. Trigger the "Zeebe General QA ITs" CI job (`ci.yml`, `integration-tests` job, `zeebe-shared-integration` group).
2. Observe `GatewayAuthenticationNoneIT` in the Maven Failsafe output.

### Current behavior
The test fails with:
```
ContainerFetchException: Can't get Docker image: RemoteDockerImage(imageName=camunda/identity:8.10.0-alpha2, ...)
Caused by: ConditionTimeoutException: Condition with Lambda expression in
    org.testcontainers.images.RemoteDockerImage was not fulfilled within 2 minutes.
```
The log shows no Docker pull output for `camunda/identity` at all — the pull stalls silently in the
Docker daemon. In the same job window, explicit `toomanyrequests` 429s appear for other `docker.io`
image pulls (e.g. `ubuntu:noble` in a concurrent Docker build step), confirming the runner's IP is
rate-limited on Docker Hub. The test eventually passes after 3–8 retries.

Observed in 4 of 5 CI runs on 2026-06-25 (run 28174664575, 28178769608, 28182346952, 28188700442).

### Expected behavior
The test passes on first attempt without requiring retries.

### Workaround
The CI retry mechanism (`FLAKY_TEST_RERUN_COUNT`) eventually succeeds. No manual action needed,
but it wastes 4–8 minutes of CI time per occurrence.

### Environment
SM

### Version
- Component Version: CI / `main` branch (identity Docker image: `camunda/identity:8.10.0-alpha2`)

### Rootcause
The `integration-tests` job in `.github/workflows/ci.yml` calls `./.github/actions/setup-build`
**without** `dockerhub-readonly: true`. The action's Docker Hub login step is guarded by that
flag, so it is skipped — all `docker.io` pulls in this job run as anonymous and are subject to
Docker Hub rate limits.

The Keycloak container never fails because `dasniko/testcontainers-keycloak` pulls from
`quay.io/keycloak/keycloak`, which has no such rate limits. Only the `camunda/identity` image
(on `docker.io`) is affected.

Sibling jobs (`rdbms-integration-tests`, `docker-checks`) already pass `dockerhub-readonly: true`
to `setup-build` and do not show this failure.

### Solution ideas
Add `dockerhub-readonly: true` to the `setup-build` call in the `integration-tests` job in
`.github/workflows/ci.yml`, consistent with what `rdbms-integration-tests` already does.

Note: the identity pull stalls silently rather than failing with an explicit 429, suggesting the
Docker daemon is starved by concurrent rate-limited pulls rather than directly rate-limited itself.
Whether adding Docker Hub auth fully resolves the flake will be confirmed by post-merge CI runs.

### Links
- Failing CI runs: https://github.com/camunda/camunda/actions/runs/28188700442, https://github.com/camunda/camunda/actions/runs/28185126267, https://github.com/camunda/camunda/actions/runs/28182346952, https://github.com/camunda/camunda/actions/runs/28178769608, https://github.com/camunda/camunda/actions/runs/28174664575
- Incident: [#inc-6283-job-ci-integration-testszeebe-shared-integration-has-high-failure](https://camunda.slack.com/archives/C0BE36QM508)

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -1104,6 +1104,7 @@ jobs:
           test_owner: "${{ matrix.owner }}"
       - uses: ./.github/actions/setup-build
         with:
+          dockerhub-readonly: true
           maven-cache-key-modifier: it-${{ matrix.group }}
           vault-address: ${{ secrets.VAULT_ADDR }}
           vault-role-id: ${{ secrets.VAULT_ROLE_ID }}

--- a/zeebe/qa/integration-tests/src/test/java/io/camunda/zeebe/it/shared/gateway/GatewayAuthenticationNoneIT.java
+++ b/zeebe/qa/integration-tests/src/test/java/io/camunda/zeebe/it/shared/gateway/GatewayAuthenticationNoneIT.java
@@ -32,7 +32,6 @@ import java.util.function.Supplier;
 import java.util.function.UnaryOperator;
 import java.util.stream.Stream;
 import org.junit.jupiter.api.BeforeEach;
-import org.junit.jupiter.api.Disabled;
 import org.junit.jupiter.api.Named;
 import org.junit.jupiter.api.extension.RegisterExtension;
 import org.junit.jupiter.params.ParameterizedTest;
@@ -51,7 +50,6 @@ import org.testcontainers.utility.DockerImageName;
  * even when the {@link Profile#CONSOLIDATED_AUTH} profile is active. In other words, users must be
  * able to override the security configuration with env vars.
  */
-@Disabled("Flaky in CI; see https://github.com/camunda/camunda/issues/56057")
 @Testcontainers
 @ZeebeIntegration
 public class GatewayAuthenticationNoneIT {

